relayer/chains/evm/types/types.go

```diff
@@ -20,39 +22,59 @@ type Block struct {
 	GasUsed      string   `json:"gasUsed"`
 }
 
+type NonceValue struct {
+	LastUpdated time.Time
+	Previous    *big.Int
+	Current     *big.Int
+}
+
 type NonceTracker struct {
-	address map[common.Address]*big.Int
+	address map[common.Address]*NonceValue
+	Getter  func(context.Context, common.Address, *big.Int) (*big.Int, error)
 	*sync.Mutex
 }
 
+var NonceUpdateInterval = 3 * time.Minute
+
 type NonceTrackerI interface {
 	Get(common.Address) *big.Int
 	Set(common.Address, *big.Int)
-	Inc(common.Address)
 }
 
 // NewNonceTracker
-func NewNonceTracker() NonceTrackerI {
+func NewNonceTracker(getter func(context.Context, common.Address, *big.Int) (*big.Int, error)) NonceTrackerI {
```

Comment:
Yeah very smart. I too have had done the same during experiments.

That would work for faster chains but not for chains like base.

Reply:
@debendraoli moving forward lets split large PR into smaller chunks so its easier to review and test. Larger the PR more time it will take to review and get a full sense of the impact.

Reply:
Yeah definitely. It was very fast paced with lots of experiments. Otherwise I try to splits into smaller task as possible.